plugins/inputs/solrmetrics/solrmetrics.go

```diff
@@ -0,0 +1,413 @@
+			"solr.node:CONTAINER.cores.lazy",
+			"solr.node:CONTAINER.cores.loaded",
+			"solr.node:CONTAINER.cores.unloaded"
+		]
+	`
+)
+
+// ResponseHeader is an exported type that
+// contains a response metrics: QTime and Status
+type ResponseHeader struct {
+	QTime  int64 `json:"QTime"`
+	Status int64 `json:"status"`
+}
+
+// Serie is type to represent series
+type Serie struct {
+	mesName   string // Measurament name
+	fields    Fields
+	tags      Tags
+	timestamp time.Time
+}
+
+// Tags for series
+type Tags map[string]string
+
+// Metrics are type to represent generic metrics returned by Solr
+type Metrics map[string]interface{}
+
+// Fields to export
+type Fields map[string]interface{}
+
+// MetricResponse is an exported type that
+// contains Metrics of different types and represents response of Solr
+type MetricResponse struct {
+	ResponseHeader ResponseHeader `json:"responseHeader"`
+	Metrics        Metrics        `json:"metrics"`
+}
+
+// Solr is a plugin to read stats from one or many Solr servers
+type Solr struct {
+	Local       bool
+	Servers     []string
+	Username    string
+	Password    string
+	Prefixes    []string
+	Keys        []string
+	Httptimeout time.Duration
+	Cores       []string
+	client      *http.Client
+	reqs        []*http.Request
+}
+
+func copyStrMaps(src, dst map[string]string) {
+	for k := range src {
+		dst[k] = src[k]
+	}
+}
+
+// parseMetric parses metric and returns fields with name "preffix.field" and values
+func parseMetrics(metrics Metrics, acc telegraf.Accumulator) Metrics {
+	f := make(Metrics)
+	for k, v := range metrics {
+		t := reflect.ValueOf(v)
+		switch t.Kind() {
+		case reflect.Int:
+			f[k] = t.Int()
+		case reflect.Float32, reflect.Float64:
+			f[k] = t.Float()
+		case reflect.String:
+			f[k] = t.String()
+		case reflect.Bool:
+			f[k] = t.Bool()
+		case reflect.Map:
+			subMetsMap, ok := v.(map[string]interface{})
+			if !ok {
+				acc.AddError(fmt.Errorf("error of converting map"))
+				continue
+			}
+			subMets := parseMetrics(subMetsMap, acc)
+			if len(subMets) == 0 {
+				continue
+			}
+			for km, vm := range subMets {
+				key := k + delimeter + km
+				f[key] = vm
+			}
+		default:
+			acc.AddError(fmt.Errorf("error of parsing, unknow type. %s=%s", k, v))
+			continue
+		}
+	}
+	return f
+}
+
+// getMetricTags parces metric name to collection, shard, replica, metric
+func getCollectionTags(metricName string) ([]string, error) {
+	ct := strings.SplitN(metricName, ".", 4)
+	if len(ct) < 4 {
+		return nil, fmt.Errorf("wrong metric name %s", metricName)
+	}
+	return ct, nil
+}
+
+func metricsToSeries(metrics Metrics, addTags Tags, acc telegraf.Accumulator) (res []Serie) {
+	mTime := time.Now()
+	series := make(map[string]Serie)
+
+	// Create series for standard groups
+	for _, group := range []string{"jvm", "jetty", "node"} {
+		series[group] = Serie{mesName: group, fields: make(Fields), tags: make(Tags), timestamp: mTime}
+		copyStrMaps(addTags, series[group].tags)
```

Comment:
`series[group].tags` should be empty shouldn't it? What is the point in this function call?